esmf_regrid/tests/unit/experimental/unstructured_scheme/test__create_cube.py

```diff
@@ -11,17 +13,56 @@ def test_create_cube_2D():
     data = np.ones([2, 3])
 
     # Create a source cube with metadata and scalar coords
-    src_cube = iris.cube.Cube(np.zeros(5))
+    src_cube = Cube(np.zeros(5))
     src_cube.units = "K"
     src_cube.attributes = {"a": 1}
     src_cube.standard_name = "air_temperature"
-    scalar_height = iris.coords.AuxCoord([5], units="m", standard_name="height")
-    scalar_time = iris.coords.DimCoord([10], units="s", standard_name="time")
+    scalar_height = AuxCoord([5], units="m", standard_name="height")
+    scalar_time = DimCoord([10], units="s", standard_name="time")
     src_cube.add_aux_coord(scalar_height)
     src_cube.add_aux_coord(scalar_time)
 
     mesh_dim = 0
 
+    grid_x = DimCoord(np.arange(3), standard_name="longitude")
+    grid_y = DimCoord(np.arange(2), standard_name="latitude")
+
+    cube = _create_cube(data, src_cube, mesh_dim, grid_x, grid_y)
+    src_metadata = src_cube.metadata
+
+    expected_cube = Cube(data)
+    expected_cube.metadata = src_metadata
+    expected_cube.add_dim_coord(grid_x, 1)
+    expected_cube.add_dim_coord(grid_y, 0)
+    expected_cube.add_aux_coord(scalar_height)
+    expected_cube.add_aux_coord(scalar_time)
+    assert expected_cube == cube
+
+
+def test_create_cube_4D():
+    """Test creation of 2D output grid."""
+    data = np.ones([4, 2, 3, 5])
+
+    # Create a source cube with metadata and scalar coords
+    src_cube = Cube(np.zeros([4, 5, 5]))
+    src_cube.units = "K"
+    src_cube.attributes = {"a": 1}
+    src_cube.standard_name = "air_temperature"
+    scalar_height = AuxCoord([5], units="m", standard_name="height")
+    scalar_time = DimCoord([10], units="s", standard_name="time")
```

Comment:
Is this right? Adding a `DimCoord` as an `add_aux_coord`?

Reply:
This is a bit of Iris weirdness. While DimCoord and AuxCoord are different classes, a cube also has two "boxes" to put coords in: `_aux_coords_and_dims` and `_dim_coords_and_dims`. Somewhat counterintuitively, `_aux_coords_and_dims` can contain DimCoords. In fact, since this is a scalar coordinate, it has to be added to `_aux_coords_and_dims` since it does not describe a dimension (all coords in `_dim_coords_and_dims` must be associated with a dimension). Scalar DimCoords will be generated when a cube is sliced, so these are worth considering and testing with.

Reply:
Great, thanks for the explanation.  With that in mind, are we comfortable these tests are sufficiently complete in their coverage for now?

Reply:
I think so, there's still some stuff like derived coordinates which we might want to add in the future, but I think this covers the main stuff.